test/metabase/upload_test.clj

```diff
@@ -1349,40 +1349,53 @@
 (deftest append-column-mismatch-test
   (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
     (with-uploads-allowed
-      (testing "Append should fail if there are extra or missing columns in the CSV file"
+      (testing "Append should fail only if there are missing columns in the CSV file"
         (doseq [[csv-rows error-message]
-                {["_mb_row_id,id,name,extra column one,EXTRA COLUMN TWO"]
-                 (trim-lines "The CSV file contains extra columns that are not in the table:
-                              - extra_column_two
-                              - extra_column_one")
-
-                 [""]
+                {[""]
                  (trim-lines "The CSV file is missing columns that are in the table:
                               - id
                               - name")
 
+                 ;; Extra columns are fine, as long as none are missing.
+                 ["_mb_row_id,id,extra 1, extra 2,name"]
+                 nil
```

Comment:
Is this test necessary when we coverage in `append-new-column-test`?

Reply:
You're right that this overlap is awkward. Since this is the only test that covers inserting multiple columns, and that turned out to be an important edge case, I'm inclined to leave for now.

Let's use the Replacement Arc as an opportunity to factor the tests better in a bunch of ways.

Reply:
yeah, maybe the test that explicitly has the title `append-new-column-test` should test inserting multiple columns? It seems more incidental that this tested multiple columns.